Automattic/wp-calypso
# Put "Installed Plugins" & "Upload" plugin buttons behind feature gates

#### Proposed Changes

Redirects the following two pages to /plugins/{site_slug}

- /plugins/upload/{site_slug} [If missing `WPCOM_FEATURES_UPLOAD_PLUGINS`]
- /plugins/manage/{site_slug} [If missing `WPCOM_FEATURES_MANAGE_PLUGINS`]

Hides the "Installed Plugins" and "Upload" buttons if the site is missing the `WPCOM_FEATURES_MANAGE_PLUGINS` or `WPCOM_FEATURES_UPLOAD_PLUGINS` respectively.

Before | After
--|--
![before-plugins](https://user-images.githubusercontent.com/140841/175096294-6337f55b-9883-4e8d-af31-9a799d23956e.png) | ![after-plugins](https://user-images.githubusercontent.com/140841/175096324-21766b73-dbb3-4eb5-9b4f-9c0d615b2807.png)

The "Upload" button on the /plugins/manage/{site_slug} is also gated by UPLOAD_PLUGINS. So if the site were to have MANAGE_PLUGINS but not UPLOAD_PLUGINS, the upload button will not appear.

Before | After
--|--
![manage-before](https://user-images.githubusercontent.com/140841/175110931-15e30142-8af9-42f2-a9b0-fd28efe57b5a.png) | ![manage-after](https://user-images.githubusercontent.com/140841/175110940-7e4a0bd3-805f-4584-aa59-32673bbed142.png)

#### Testing Instructions

> **Note**
> Jetpack (non-Atomic) sites should always show the "Installed Plugins" & "Upload" plugin buttons, and the /plugins/upload/{site_slug} and /plugins/manage/{site_slug} pages should NOT redirect.

* Load this branch.
* Got to /plugins/{site_slug} on a sub Business/Pro plan.
  * You should NOT see the "Installed Plugins" or "Upload" buttons.
* Try manually going to /plugins/upload/{site_slug} & /plugins/manage/{site_slug}
  * You should be redirected to /plugins/{site_slug}
* Got to /plugins/{site_slug} on a Business/Pro or higher plan.
  * You should see the "Installed Plugins" and "Upload" buttons.
* Try manually going to /plugins/upload/{site_slug} & /plugins/manage/{site_slug}
  * You should NOT be redirected to /plugins/{site_slug}

This PR is related to #64116
